mnist/README.md

```diff
@@ -6,165 +6,96 @@ This example guides you through the process of taking an example model, modifyin
-- Contain 3 nodes of at least 8 cores and 16 GB of RAM.
-
-If using GKE, the following will provision a cluster with the required features:
-
-```
-export CLOUDSDK_CONTAINER_USE_CLIENT_CERTIFICATE=True
-gcloud alpha container clusters create ${USER} --enable-kubernetes-alpha --machine-type=n1-standard-8 --num-nodes=3 --disk-size=200 --zone=us-west1-a --cluster-version=1.9.3-gke.0 --image-type=UBUNTU
-```
-
-If using Azure, the following will provision a cluster with the required features, [using the az cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest):
+## 2.ksonnect cli
 
 ```
-# Create a resource group
-az group create -n kubeflowrg -l eastus
-# Deploy the cluster
-az aks create -n kubeflowaks -g kubeflowrg -l eastus -k 1.9.6 -c 3 -s Standard_NC6
-# Authentication into the cluster
-az aks get-credentials -n kubeflowaks -g kubeflowrg
+curl -fksSL https://github.com/ksonnet/ksonnet/releases/download/v0.13.0/ks_0.13.0_linux_amd64.tar.gz \
+    | tar --strip-components=1 -xvz -C /usr/local/bin/ ks_0.13.0_linux_amd64/ks
 ```
 
-NOTE: You must be a Kubernetes admin to follow this guide. If you are not an admin, please contact your local cluster administrator for a client cert, or credentials to pass into the following commands:
+## 3.Kubeflow environment
 
 ```
-$ kubectl config set-credentials <username> --username=<admin_username> --password=<admin_password>
-$ kubectl config set-context <context_name> --cluster=<cluster_name> --user=<username> --namespace=<namespace>
-$ kubectl config use-context <context_name>
+git clone -b v0.3.2 https://github.com/kubeflow/kubeflow
+./kubeflow/scripts/kfctl.sh init kf-app --platform none
+cd kf-app
+../kubeflow/scripts/kfctl.sh generate k8s
+../kubeflow/scripts/kfctl.sh apply k8s
 ```
 
-### Local Setup
+For detail info, check here: https://www.kubeflow.org/docs/started/getting-started/
 
-You also need the following command line tools:
+Note: The vizier-db in Kubeflow depends on a PV, you need to create it in order to make vizier-db work. But this does not affect this example.
 
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
-- [argo](https://github.com/argoproj/argo/blob/master/demo.md#1-download-argo)
-- [ksonnet](https://ksonnet.io/#get-started)
 
-To run the client at the end of the example, you must have [requirements.txt](requirements.txt) intalled in your active python environment.
+## 4.Setup Minio
 
-```
-pip install -r requirements.txt
-```
+Minio provides an S3 compatible API, we will use Minio in this example to act as S3.
 
-NOTE: These instructions rely on Github, and may cause issues if behind a firewall with many Github users. Make sure you [generate and export this token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/):
+### Deploy Minio
 
 ```
-export GITHUB_TOKEN=xxxxxxxx
+docker run -e MINIO_ACCESS_KEY=minio -e MINIO_SECRET_KEY=minio123 \
```

Comment:
Are you running minio locally? Shouldn't you be deploying it on the cluster? How are pods running in the cluster supposed to access a locally running docker container?

Reply:
Running in the cluster need PV to persist the data, usually in demo environment there is no out-of-box PV.

This guide is for newbies, I want to ensure they can follow the steps one by one to run this example. For skilled person, they already know how to tweak the steps to use their own solution, such as they may you Minio helm chart to deploy their own Minio.

Reply:
I agree we should target newbies. So why introduce MinIO at all? If folks are running on a cloud they should be able to just use S3,GCS or HDFS directly (TF natively supports those filesystems).

If folks are running on prem and don't have an object store, why not just have them run non-distributed?